### DISCUSSION TITLE
How should I deal with duplicates in v3?

### DISCUSSION THREAD

**Discussion by maxrzaw** [folke/trouble.nvim#538] [Q&A]
I have been trying out v3 and I have found a couple scenarios with the new `lsp_definitions` that I don't know how to deal with. The auto jump feature does not work like it did in v2 and I am getting duplicate values. The first scenario I noticed was with `lua_ls`. A minimal reproducible file is
```lua
local test_function = function()
    print("Hello, World!")
end

test_function()
```
when running `:Trouble lsp_definitions toggle<cr>` on the last line, it does not auto jump and instead shows two items in trouble.
```
  Definitions  2 
   └╴  test.lua  2 
     ├╴local test_function = function() (lua_ls) [1, 7]
     └╴local test_function = function() (lua_ls) [1, 23]
```

Is there a way to make this auto jump?

<details>
<summary>My keybinding and trouble config</summary>

Trouble config:
```lua
        trouble.setup({
            auto_close = true,
            focus = true,
            modes = {
                lsp_references = {
                    params = {
                        include_declaration = false,
                    },
                },
                lsp_base = {
                    params = {
                        include_current = false,
                    },
                },
            },
            auto_jump = {
                "lsp",
                "lsp_definitions",
                "lsp_type_definitions",
            },
        })
```
Keybinding:
```lua
            vim.keymap.set(
                "n",
                "gd",
                "<cmd>Trouble lsp_definitions toggle<cr>",
                vim.tbl_extend("error", opts, { desc = "LSP: [G]o to [D]efinitions" })
            )

```
</details>

**Comment by disrupted** [comment]
I've solved it by applying a custom filter https://github.com/disrupted/dotfiles/commit/a76952c3be31b57a0cabfa03efb118101b285878

there is also an open issue for Lua LS specifically https://github.com/LuaLS/lua-language-server/issues/2451
